packages/did-core-test-server/suites/did-consumption/did-json-consumption.js

```diff
@@ -0,0 +1,133 @@
+          'set by converting the JSON Object member value according to the ' +
+          'JSON representation type as defined in this table. Since order ' +
+          'is not specified by JSON Objects, no insertion order is ' +
+          'guaranteed.', () => {
+          expect(reserializationSuccess(value)).toBe(true);
+        });
+      } else if(typeof value === 'string' && isXmlDatetime(value)) {
+        it('6.2.2 JSON Consumption - JSON String where data model entry ' +
+          'value is a datetime: A datetime.', () => {
+          expect(reserializationSuccess(value)).toBe(true);
+        });
+      } else if(typeof value === 'string') {
+        it('6.2.2 JSON Consumption - JSON String, where the data model ' +
+          'entry value type is string or unknown: A string.', () => {
+          expect(reserializationSuccess(value)).toBe(true);
+        });
+      } else if(typeof value === 'number') {
+        if(!Number.isInteger(value)) {
+          it('6.2.2 JSON Consumption - JSON Number without a decimal or ' +
+            'fractional component: An integer.', () => {
+            expect(reserializationSuccess(value)).toBe(true);
+          });
+        } else {
+          it('6.2.2 JSON Consumption - JSON Number with a decimal and ' +
+            'fractional component, or when entry value is a double ' +
+            'regardless of inclusion of fractional component: ' +
+            'A double.', () => {
+            expect(reserializationSuccess(value)).toBe(true);
+          });
+        }
+      } else if(typeof value === 'boolean') {
+        it('6.2.2 JSON Consumption - JSON Boolean: A boolean.', () => {
+          expect(reserializationSuccess(value)).toBe(true);
+        });
+      } else if(value === null) {
+        it('6.2.2 JSON Consumption - JSON null literal: A null value.', () => {
+          expect(reserializationSuccess(value)).toBe(true);
+        });
+      } else {
+        it('UNKNOWN JSON TYPE '+ value, () => {
+          expect(true).toBe(true);
+        });
+      }
+    });
+  });
+
+  it('6.2.2 JSON Consumption - A conforming consumer for a JSON ' +
+    'representation that is consuming a DID document with a root ' +
+    'element that is not a JSON Object MUST report an error.', () => {
+    expect(typeof didDocument === 'object');
+  });
+
+  it('6.2.2 JSON Production - If media type information is available to ' +
+    'a conforming consumer and the media type value is ' +
+    'application/did+json, then the data structure being consumed is a ' +
+    'DID document, and the root element MUST be a JSON Object where all ' +
+    'members of the object are entries of the DID document.', async () => {
+      const {contentType} = resolutionResult.didResolutionMetadata;
+      expect(contentType).toBe('application/did+json');
+  });
+}
+
+const reserializationSuccess = (value) => {
+  const reserialization = JSON.parse(JSON.stringify(value));
```

Comment:
Is it safer to wrap for an exception, even we know its structure is well-formed?  e.g.,  https://github.com/w3c/did-test-suite/blob/385760b56a7627294bfff04831e63887a0101b87/packages/did-core-test-server/suites/did-core-properties/did-metadata-structure.js#L42-L51

Reply:
I looked into this. Ideally, we'd use a Jest test for this, but I couldn't figure out exactly what to write. I think it's fine for now. We can always refactor later. If an exception is thrown, Jest will say where since we're in a test when it happens.